lxd/instance/drivers/driver_qemu.go

```diff
@@ -2888,6 +2882,17 @@ func (d *qemu) generateQemuConfigFile(cpuInfo *cpuTopology, mountInfo *storagePo
 			return "", nil, fmt.Errorf("Unable to locate matching firmware: %+v", firmwares)
 		}
 
+		// As 2MB firmware was deprecated in the LXD snap we have to regenerate NVRAM for VMs which used the 2MB one.
+		if shared.InSnap() && !strings.Contains(ovmfCode, "4MB") {
```

Comment:
Should we use `ovmfGenericFirmwares[0].code` here instead of 4MB?

Reply:
> ovmfGenericFirmwares[0].code

nope, because ovmfCode can be `OVMF_CODE.2MB.fd`, `OVMF_CODE.2MB.CSM.fd`, `OVMF_CODE.fd` or `OVMF_CODE.CSM.fd`. So we don't expect that `ovmfCode` will be `ovmfGenericFirmwares[0].code` after the "forcing" 4MB firmware, it's perfectly fine to have `ovmfCSMFirmwares[0].code` too as it's also 4 meg thing.